django_mongodb/query_utils.py

```diff
@@ -48,4 +48,6 @@ def process_rhs(node, compiler, connection):
 def regex_match(field, value, regex, *re_args, **re_kwargs):
     regex = re.compile(regex % re.escape(value), *re_args, **re_kwargs)
     options = "i" if regex.flags & re.I else ""
-    return {"$regexMatch": {"input": field, "regex": regex.pattern, "options": options}}
+    return {
+        "$regexMatch": {"input": {"$toString": field}, "regex": regex.pattern, "options": options}
```

Comment:
Does this change end up solving anything for JSONField? I reverted and `model_fields.test_jsonfield` still passed. Maybe there's another affected test. Anyway, I'd make it a separate PR since it fixes other non-JSONField tests.

Reply:
No, it doesn't. It solve only when a json field is a string, number, or null. But It does not pass any new test.